topic_monitor/scripts/data_publisher.py

```diff
@@ -0,0 +1,125 @@
+        '--transient-local', action='store_true', default=False,
+        help='Set QoS durability option to "transient local"')
+
+    parser.add_argument(
+        '--depth', type=int, default=default_depth, help='Size of the QoS history depth')
+
+    parser.add_argument(
+        '--keep-all', action='store_true', default=False,
+        help='Set QoS history option to "keep all" (unlimited depth, subject to resource limits)')
+
+    parser.add_argument(
+        '--payload-size', type=int, default=0, help='Size of data payload to send')
+
+    parser.add_argument(
+        '--period', type=float, default=0.5, help='Time in seconds between messages')
+
+    parser.add_argument(
+        '--end-after', type=int, help='Script will exit after publishing this many messages')
+
+    args = parser.parse_args()
+
+    qos_profile = QoSProfile()
+
+    if args.best_effort:
+        print('Reliability: best effort')
+        qos_profile.reliability = QoSReliabilityPolicy.RMW_QOS_POLICY_BEST_EFFORT
+    else:
+        print('Reliability: reliable')
+        qos_profile.reliability = QoSReliabilityPolicy.RMW_QOS_POLICY_RELIABLE
+
+    if args.keep_all:
+        print('History: keep all')
+        qos_profile.history = QoSHistoryPolicy.RMW_QOS_POLICY_KEEP_ALL_HISTORY
+    else:
+        print('History: keep last')
+        qos_profile.history = QoSHistoryPolicy.RMW_QOS_POLICY_KEEP_LAST_HISTORY
+
+    print('Depth: {0}'.format(args.depth))
+    qos_profile.depth = args.depth
+
+    if args.transient_local:
+        print('Durability: transient local')
+        qos_profile.durability = QoSDurabilityPolicy.RMW_QOS_POLICY_TRANSIENT_LOCAL_DURABILITY
+    else:
+        print('Durability: volatile')
+        qos_profile.durability = QoSDurabilityPolicy.RMW_QOS_POLICY_VOLATILE_DURABILITY
+
+    print('Payload size: {0}'.format(args.payload_size))
+    data = 'a' * args.payload_size
+
+    reliability_suffix = '_best_effort' if args.best_effort else ''
+    topic_name = '{0}_data{1}'.format(args.data_name, reliability_suffix)
+    print('Publishing on topic: {0}'.format(topic_name))
+    sys.stdout.flush()
+
+    rclpy.init()
+    node = rclpy.create_node('%s_pub' % topic_name)
+    data_pub = node.create_publisher(Header, topic_name, qos_profile)
+
+    msg = Header()
+    cycle_count = 0
+
+    def publish_msg(val):
+        msg.frame_id = '{0}_{1}'.format(val, data)
+        data_pub.publish(msg)
+        print('Publishing: "{0}"'.format(val))
+        sys.stdout.flush()  # this is to get the output to show immediately when using Launch
+
+    while rclpy.ok():
+        if args.end_after is not None and cycle_count >= args.end_after:
+            publish_msg(-1)
+            sleep(0.1)  # allow reliable messages to get re-sent if needed
+            return
+
+        publish_msg(cycle_count)
+        cycle_count += 1
+
+        try:
+            sleep(args.period)
+        except KeyboardInterrupt:
+            publish_msg(-1)
+            sleep(0.1)
```

Comment:
destroying the node and calling shutdown here would avoid ghost DDS entities (like you did in https://github.com/ros2/demos/pull/90/files#diff-00d7c7eb56daa7f13e18b965d5a3572cR296)